apps/emqx_bridge_pulsar/src/emqx_bridge_pulsar.erl

```diff
@@ -0,0 +1,228 @@
+%%--------------------------------------------------------------------
+-module(emqx_bridge_pulsar).
+
+-include("emqx_bridge_pulsar.hrl").
+-include_lib("emqx_connector/include/emqx_connector.hrl").
+-include_lib("typerefl/include/types.hrl").
+-include_lib("hocon/include/hoconsc.hrl").
+
+%% hocon_schema API
+-export([
+    namespace/0,
+    roots/0,
+    fields/1,
+    desc/1
+]).
+%% emqx_ee_bridge "unofficial" API
+-export([conn_bridge_examples/1]).
+
+%%-------------------------------------------------------------------------------------------------
+%% `hocon_schema' API
+%%-------------------------------------------------------------------------------------------------
+
+namespace() ->
+    "bridge_pulsar".
+
+roots() ->
+    [].
+
+fields(pulsar_producer) ->
+    fields(config) ++ fields(producer_opts);
+fields(config) ->
+    [
+        {enable, mk(boolean(), #{desc => ?DESC("config_enable"), default => true})},
+        {servers,
+            mk(
+                binary(),
+                #{
+                    required => true,
+                    desc => ?DESC("servers"),
+                    validator => emqx_schema:servers_validator(
+                        ?PULSAR_HOST_OPTIONS, _Required = true
+                    )
+                }
+            )},
+        {authentication,
+            mk(hoconsc:union([none, ref(auth_basic), ref(auth_token)]), #{
+                default => none, desc => ?DESC("authentication")
+            })}
+    ] ++ emqx_connector_schema_lib:ssl_fields();
+fields(producer_opts) ->
+    [
+        {batch_size,
+            mk(
+                pos_integer(),
+                #{default => 100, desc => ?DESC("producer_batch_size")}
+            )},
+        {compression,
+            mk(
+                hoconsc:enum([no_compression, snappy, zlib]),
+                #{default => no_compression, desc => ?DESC("producer_compression")}
+            )},
+        {send_buffer,
+            mk(emqx_schema:bytesize(), #{
+                default => <<"1MB">>, desc => ?DESC("producer_send_buffer")
+            })},
+        {sync_timeout,
+            mk(emqx_schema:duration_ms(), #{
+                default => <<"3s">>, desc => ?DESC("producer_sync_timeout")
+            })},
+        {retention_period,
+            mk(
+                hoconsc:union([infinity, emqx_schema:duration_ms()]),
+                #{default => infinity, desc => ?DESC("producer_retention_period")}
+            )},
+        {max_batch_bytes,
+            mk(
+                emqx_schema:bytesize(),
+                #{default => <<"900KB">>, desc => ?DESC("producer_max_batch_bytes")}
+            )},
+        {local_topic, mk(binary(), #{required => false, desc => ?DESC("producer_local_topic")})},
+        {pulsar_topic, mk(binary(), #{required => true, desc => ?DESC("producer_pulsar_topic")})},
+        {strategy,
+            mk(
+                hoconsc:enum([random, roundrobin, first_key_dispatch]),
```

Comment:
keep it the same as Kafka, `key_dispatch` from the config, but translate to `first_key_dispath` at runtime.
```suggestion
                hoconsc:enum([random, roundrobin, key_dispatch]),
```